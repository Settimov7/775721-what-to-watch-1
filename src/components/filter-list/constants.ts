export const GenreToFilterTitle = {
  all: `All genres`,
  Crime: `Crime`,
  Action: `Action`,
  Drama: `Drama`,
  Comedy: `Comedies`,
  Adventure: `Adventure`,
  Fantasy: `Fantasy`,
  Thriller: `Thriller`,
};
export const MAX_FILTERS = 10;
