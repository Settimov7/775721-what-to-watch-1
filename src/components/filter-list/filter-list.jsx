import * as React from 'react';
import * as PropTypes from 'prop-types';

const GenreToFilterTitle = {
  all: `All genres`,
  Crime: `Crime`,
  Action: `Action`,
  Drama: `Drama`,
  Comedy: `Comedies`,
  Adventure: `Adventure`,
  Fantasy: `Fantasy`,
  Thriller: `Thriller`,
};

export const FilterList = (props) => {
  const {properties, currentFilter, changeCurrentFilter} = props;
  const filters = [`all`, ...new Set(properties)];

  return (
    <ul className="catalog__genres-list">
      {filters.map((filter) => {
        const isActive = filter === currentFilter;

        return (
          <li
            key={filter}
            className={`catalog__genres-item ${isActive && `catalog__genres-item--active`}`}
          >
            <a
              href="#"
              className="catalog__genres-link"
              onClick={(evt) => {
                evt.preventDefault();

                if (!isActive) {
                  changeCurrentFilter(filter);
                }
              }}
            >
              {GenreToFilterTitle[filter]}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

FilterList.propTypes = {
  properties: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentFilter: PropTypes.string.isRequired,
  changeCurrentFilter: PropTypes.func.isRequired,
};

