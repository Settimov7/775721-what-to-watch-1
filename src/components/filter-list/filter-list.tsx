import * as React from 'react';
import {connect} from 'react-redux';

import {ActionCreator as FilterActionCreator} from "../../reducer/filter/filter";
import {ActionCreator as FilmsActionCreator} from '../../reducer/films/films';
import {getFilmsGenres} from "../../reducer/films/selectors";
import {getCurrentFilter} from "../../reducer/filter/selectors";

const GenreToFilterTitle = {
  all: `All genres`,
  Crime: `Crime`,
  Action: `Action`,
  Drama: `Drama`,
  Comedy: `Comedies`,
  Adventure: `Adventure`,
  Fantasy: `Fantasy`,
  Thriller: `Thriller`,
};

interface Props {
  properties: string[],
  activeItem: string,
  onChange: (genre: string) => void,
}

export const FilterList = (props: Props) => {
  const {properties, activeItem, onChange} = props;
  const filters = [`all`, ...new Set(properties)];

  return (
    <ul className="catalog__genres-list">
      {filters.map((filter) => {
        const isActive = filter === activeItem;

        return (
          <li
            key={filter}
            className={`catalog__genres-item ${isActive && `catalog__genres-item--active`}`}
          >
            <a
              href="#"
              className="catalog__genres-link"
              onClick={(evt) => {
                evt.preventDefault();

                if (!isActive) {
                  onChange(filter);
                }
              }}
            >
              {GenreToFilterTitle[filter]}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  properties: getFilmsGenres(state),
  activeItem: getCurrentFilter(state),
});

const mapDispatchToProps = (dispatch) => ({
  onChange: (genre: string): void => {
    dispatch(FilterActionCreator.changeCurrentFilterByFilmGenre(genre));
    dispatch(FilmsActionCreator.resetDisplayedFilmsNumber());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(FilterList);

