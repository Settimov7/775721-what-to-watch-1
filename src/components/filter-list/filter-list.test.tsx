import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {FilterList} from "./filter-list";

it(`Filters list correctly renders`, () => {
  const props = {
    properties: [`Filter-1`, `Filter-2`, `Filter-3`, `Filter-4`, `Filter-5`, `Filter-6`, `Filter-7`, `Filter-8`, `Filter-9`, `Filter-10`, `Filter-11`],
    activeItem: `Filter-1`,
    onChange: jest.fn(),
  };

  const tree = renderer.create(<FilterList {...props}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
