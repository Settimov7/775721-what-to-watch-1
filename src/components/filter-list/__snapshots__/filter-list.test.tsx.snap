// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters list correctly renders 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-1
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-2
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-3
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-4
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-5
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-6
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-7
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-8
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Filter-9
    </a>
  </li>
</ul>
`;
